internal/codespaces/codespaces.go

```diff
@@ -37,33 +37,37 @@ type logger interface {
 
 // ConnectToLiveshare waits for a Codespace to become running,
 // and connects to it using a Live Share session.
-func ConnectToLiveshare(ctx context.Context, progress progressIndicator, sessionLogger logger, apiClient apiClient, codespace *api.Codespace) (sess *liveshare.Session, err error) {
+func ConnectToLiveshare(ctx context.Context, progress progressIndicator, sessionLogger logger, apiClient apiClient, codespace *api.Codespace) (*liveshare.Session, error) {
 	if codespace.State != api.CodespaceStateAvailable {
 		progress.StartProgressIndicatorWithLabel("Starting codespace")
 		defer progress.StopProgressIndicator()
 		if err := apiClient.StartCodespace(ctx, codespace.Name); err != nil {
 			return nil, fmt.Errorf("error starting codespace: %w", err)
 		}
 	}
-	expBackoff := backoff.NewExponentialBackOff()
 
-	expBackoff.Multiplier = 1.1
-	expBackoff.MaxInterval = 10 * time.Second
-	expBackoff.MaxElapsedTime = 5 * time.Minute
-
-	for retries := 0; !connectionReady(codespace); retries++ {
-		if retries > 1 {
-			duration := expBackoff.NextBackOff()
-			time.Sleep(duration)
-		}
-
-		if expBackoff.GetElapsedTime() >= expBackoff.MaxElapsedTime {
-			return nil, errors.New("timed out while waiting for the codespace to start")
-		}
-
-		codespace, err = apiClient.GetCodespace(ctx, codespace.Name, true)
+	if !connectionReady(codespace) {
```

Comment:
Nit: can we put in some whitespace to break this up a bit? It's pretty dense to read :smile: